File: ProductManagement/Client-Side/src/app/viewModels/product-management-viewModels/product-management-viewModels/response/response-get-categories-viewModel.ts
export class ResponseGetCategoriesViewModel {
    constructor(categories) {
      this.categories = categories;
    }

    public categories: Array<ResponseGetCategoriesViewModelItem>;
}

export class ResponseGetCategoriesViewModelItem {
    constructor(id, name) {
      this.id = id;
      this.name = name;
    }
  
    public id: number;
    public name: string;
}

File: ProductManagement.ViewModels/ProductManagementViewModels/Response/ResponseGetCategoriesViewModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProductManagement.ViewModels.ProductManagementViewModels.Response
{
    public class ResponseGetCategoriesViewModel
    {
        public List<ResponseGetCategoriesViewModelItem> Categories { get; set; }

        public ResponseGetCategoriesViewModel()
        {
            Categories = new List<ResponseGetCategoriesViewModelItem>();
        }
    }

    public class ResponseGetCategoriesViewModelItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }
}
